### ISSUE TITLE
[A11y] Navigation confusing for Screenreaders

### ISSUE THREAD

**Issue by liviakuenzli** [lyne-design-system/lyne-components#1595]
## Detailed description

According to our first accessibility review by Mürsel Genc, the whole is always read fully (one after the other) when opening the navigation. Instead, only the focused element should be output and the user should navigate the elements using focus.

![image](https://user-images.githubusercontent.com/7348876/217015442-9421b9d1-244c-4deb-9169-c76f3837d7fa.png)

See also: https://flow.sbb.ch/browse/SBBCH-15183

Note by @kyubisation: Evaluate whether we really should use the `<dialog>` element for `<sbb-navigation>`, as this might be what is causing this behavior.
